src/eval_context.rs

```diff
@@ -1521,8 +1527,9 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
                 //let src = adt::MaybeSizedValue::sized(src);
                 //let dst = adt::MaybeSizedValue::sized(dst);
                 let src_ptr = match src {
-                    Value::ByRef(ptr) => ptr,
-                    _ => bug!("expected pointer, got {:?}", src),
+                    Value::ByRef(ptr, true) => ptr,
+                    // TODO: Is it possible for unaligned pointers to occur here?
```

Comment:
Sry I can't get that to typecheck.  TBH I never know what I am doing when dealing with unsized types and these unsizing coercions, I always randomly mutate code until the compiler accepts it, which I fail to do here.